provisioning/transport/http/devdoc/http_requirements.md

```diff
@@ -72,14 +68,14 @@ The `constructor` creates an Http transport object used to communicate with the
 
 **SRS_NODE_PROVISIONING_HTTP_18_008: [** If `forceRegistration` is specified, the registration request shall include this as a query string value named 'forceRegistration' **]**
 
-**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' **]**
+**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall PUT the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}/register' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for the response to the PUT request. **]**
 
-**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for the response to the POST request. **]**
+**SRS_NODE_PROVISIONING_HTTP_18_011: [** If the registration request times out, `register` shall call the `callback` with with the lower level error **]**
```

Comment:
>with the lower level [](start = 122, length = 20)

if the error is fed directly to the transport-agnostic client (caller?) then the error should not be specific to the transport (the transport layer has the necessary knowledge to convert transport-specific to agnostic client). If it is necessary to also forward the internal error (for debugging purposes for example) i'd encourage making it a _field of the resulting transport-agnostic error

Reply:
Agreed.  I will flush this out when I add another transport and create an official interface.

---
In reply to: [142267085](https://github.com/Azure/azure-iot-sdk-node/pull/99#discussion_r142267085) [](ancestors = 142267085)